project_generator/tools/uvision.py

```diff
@@ -69,6 +71,10 @@ class uVisionDefinitions():
             'Utilities': {
                 'Flash2': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
             },
+            'OptxFile' : {
```

Comment:
Do you mean something like this?

```
'st-link': {
    'uvproj' : {
        'TargetDlls': {
            'Driver': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
        },
        'Utilities': {
            'Flash2': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
        },
    }
    'uvopt' : {   (or maybe better 'uvoptx' because it is used only for Keil uvoptx files)
        'DebugOpt' : {
            'nTsel' : '11',
            'pMon': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
        },
        'SetRegEntry' : {
            'Key' : 'ST-LINKIII-KEIL_SWO',
        },
    },
}
```

```
uvoptx_dic['ProjectOpt']['Target']['TargetOption']['DebugOpt']['nTsel'] = self.definitions.debuggers[debugger_name]['uvopt']['DebugOpt']['nTsel']
uvoptx_dic['ProjectOpt']['Target']['TargetOption']['DebugOpt']['pMon'] = self.definitions.debuggers[debugger_name]['uvopt']['DebugOpt']['pMon']
uvoptx_dic['ProjectOpt']['Target']['TargetOption']['TargetDriverDllRegistry']['SetRegEntry']['Key'] = self.definitions.debuggers[debugger_name]['uvopt']['SetRegEntry']['Key']
```